src/community/vsi/src/main/java/org/geoserver/vsi/VSIProperties.java

```diff
@@ -0,0 +1,81 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+
+package org.geoserver.vsi;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Properties;
+import org.apache.log4j.Logger;
+import org.gdal.gdal.gdal;
+
+/**
+ * GDAL config loader
+ *
+ * @author Matthew Northcott <matthewnorthcott@catalyst.net.nz>
+ */
+public final class VSIProperties {
+
+    // Name of property that defines where the vsi.properties file is
+    public static final String LOCATION_PROPERTY = "vsi.properties.location";
+
+    private static final Logger LOGGER = Logger.getLogger(VSIProperties.class.getName());
+
+    /**
+     * Read the vsi.properties file for credentials.
+     *
+     * @return Validated Properties object of key-value pairs
+     * @throws IOException
+     */
+    public static Properties getProperties() throws IOException {
+        final String propLocation = System.getProperty(LOCATION_PROPERTY);
+
+        if (propLocation == null) {
+            throw new IOException("The system property 'vsi.properties.location' is not set.");
+        }
+
+        final Properties prop = new Properties();
+
+        FileInputStream inStream = null;
+        try {
```

Comment:
you may want to use try-with-resources since FileInputStream implements AutoCloseable